### PR TITLE
consolidate changes from various forks

### PR DESCRIPTION
**Pull request by pbauer**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -41,8 +41,10 @@ def read(*pathnames):
       install_requires=[
           'setuptools',
           'zc.queue >= 1.3',
-          'requests < 1.0.0',
+          'requests',
           'plone.z3cform',
+          'plone.api',
```
Comment (malthe) [inline setup.py]: What's the value added of depending on `plone.api`. Yes, some community members seem to favor yet another indirection, but Plone actually does have an API and `getToolByName` is part of it.

Or via acquisition, simply `self.context.<some_tool>`.

File: src/collective/linkcheck/browser.py [hunk 2]
```diff
@@ -37,3 +47,90 @@ def queue_ratio(self):
...
+        path = actual_url[len(base_url):]
+        """
+        path = '/' + '/'.join(context.getPhysicalPath()[2:])
+        actual_url = context.absolute_url()
+        tool.update(path, status)
+
+        # for now we generate body by calling the object
+        try:
+            body = context()
+        except Exception:
```
Comment (malthe) [inline src/collective/linkcheck/browser.py]: You can omit `Exception` here.

File: src/collective/linkcheck/processor.py [hunk 3]
```diff
@@ -1,24 +1,127 @@
...
+import threading
+import time
+import transaction
+
+
+def publish_module(module_name,
+                   stdin=sys.stdin, stdout=sys.stdout, stderr=sys.stderr,
+                   environ=os.environ, debug=0, request=None, response=None,
+                   extra={}):
+    """ Adapted from from ZPublisher.Test.publish_module:
```
Comment (malthe) [inline src/collective/linkcheck/processor.py]: What's the reasoning behind this function appearing here?

**Comment by malthe** [comment]
I didn't really get the `publish_module` thing but otherwise the changes seem okay to me.

**Comment by pbauer** [comment]
@malthe thanks for looking at it. I did not really get the publish_module too but I do not use it to test internal links anyway.

**Pull request merged**